reconcile/terraform_aws_route53.py

```diff
@@ -116,6 +119,50 @@ def build_desired_state(zones):
                 logging.debug(msg)
                 record["records"] = record_values
 
+            # Process '_target_namespace_zone'
+            target_namespace_zone = record.pop("_target_namespace_zone", None)
+            if target_namespace_zone:
+                tf_resources = target_namespace_zone["namespace"]["terraformResources"]
+                tf_zone_name = target_namespace_zone["name"]
+                tf_zone_resources = [
+                    tfr
+                    for tfr in tf_resources
+                    if tfr["provider"] == "route53-zone" and tfr["name"] == tf_zone_name
+                ]
+                if not tf_zone_resources:
+                    logging.error(
+                        f"{zone_name}: field `_target_namespace_zone` found "
+                        f"for record {record_name}, but target zone not found: "
+                        f"{tf_zone_name}"
+                    )
+                    sys.exit(ExitCodes.ERROR)
+                tf_zone_resource = tf_zone_resources[0]
+                tf_zone_account_name = tf_zone_resource["account"]
+                zone_account = [
+                    a for a in all_accounts if a["name"] == tf_zone_account_name
+                ][0]
+                awsapi = AWSApi(1, [zone_account], settings=settings, init_users=False)
```

Comment:
Not sure if its feasible, but this function is quite humongous. Would it make sense to separate this parsing logic into its own function? That might also help in the unit testing later.

```
@dataclass
class ZoneAccount:
    aws_api: AWSApi
    name: str
    account_name: str
    region: str
    default_region: str

def _parse_zones(raw_target_namespace_zone: Mapping[str, Any]) -> ZoneAccount:
    pass
```

Reply:
since this is relevant to the entire integration, would it make sense to keep the same style for this PR and create a ticket to refactor?

Reply:
The only concern here is that in my experience dedicated refactor tickets/PRs never come. Usually stuff then just grows over time to a point where any refactor will be a huge amount of work (looking at terrascript_client.py for instance :eyes: ). Personally, I feel like every feature PR's goal should also be to (slightly) improve the code it affects (if possible and reasonable in given time). That's why Im always in favor of making small enhancements as you touch a portion of code - and annoying people with it :sweat_smile: E.g., in this case splitting into functions would also help to accommodate for some tests.

That being said, thank you for bearing with me :) I know I can be annoying with this stuff and I might as well be too idealistic about this. This effort might not be worth it now. It would be nice though to really have a ticket in the backlog then to take a look at this refactoring situation. I am also happy to pick it up :)

Reply:
i've created https://issues.redhat.com/browse/APPSRE-4720

i'm also happy to pick it up!